ctapipe_io_magic/__init__.py

```diff
@@ -1345,172 +1267,124 @@ def load_events(uproot_file, is_mc, n_camera_pixels):
+            # The length of the pixel array is 1183, but here only the first part of the pixel
+            # information are extracted (i.e., for the current MAGIC camera geometry, the pixels
+            # between 0 and 1039 are extracted, since the other part of pixels has only zeros):
+            calib_data[event_type]['image'] = np.array(common_info['MCerPhotEvt.fPixels.fPhot'].tolist())[:, :self.n_cam_pixels]
+            calib_data[event_type]['peak_time'] = np.array(common_info['MArrivalTime.fData'].tolist())[:, :self.n_cam_pixels]
 
-        events_cut = {
-            "cosmics_stereo_events": cosmics_stereo_cut,
-            "pedestal_events": pedestal_cut,
-        }
+            if self.is_mc:
 
-        for event_type in event_types:
+                # Reading the MC information:
+                mc_info = self.uproot_file['Events'].arrays(
+                    expressions=mc_branches,
+                    cut=events_cut[event_type],
+                    library='np',
+                )
 
-            if event_type == "pedestal_events" and is_mc:
-                continue
+                # Note that the branch "MMcEvt.fPhi" seems to be the angle between the direction
+                # of the magnetic north and the momentum of a simulated primary particle, defined
+                # between -pi and pi, negative if the momentum pointing eastward, positive westward.
+                # The conversion to azimuth should be 180 - fPhi + magnetic_declination:
+                calib_data[event_type]['mc_energy'] = u.Quantity(mc_info['MMcEvt.fEnergy'], u.GeV)
+                calib_data[event_type]['mc_theta'] = u.Quantity(mc_info['MMcEvt.fTheta'], u.rad)
+                calib_data[event_type]['mc_phi'] = u.Quantity(mc_info['MMcEvt.fPhi'], u.rad)
+                calib_data[event_type]['mc_shower_primary_id'] = np.array(mc_info['MMcEvt.fPartId'], dtype=int)
+                calib_data[event_type]['mc_h_first_int'] = u.Quantity(mc_info['MMcEvt.fZFirstInteraction'], u.cm)
+                calib_data[event_type]['mc_core_x'] = u.Quantity(mc_info['MMcEvt.fCoreX'], u.cm)
+                calib_data[event_type]['mc_core_y'] = u.Quantity(mc_info['MMcEvt.fCoreY'], u.cm)
+
+                # Reading the telescope pointing information:
+                pointing = self.uproot_file['Events'].arrays(
+                    expressions=pointing_branches,
+                    cut=events_cut[event_type],
+                    library='np',
+                )
 
-            events = input_file['Events'].arrays(
-                expressions=evt_common_list,
-                cut=events_cut[event_type],
-                library="np"
-            )
+                # ToBeChecked: it seems the *.fDev* and *.fRa/fDec branches of MC calibrated
+                # file contain only zeros. If it is the case for all MC files, the following
+                # calculations do not needed and can be simplified:
```

Comment:
I would leave it, since when real star data are read, this part of code will be used

Reply:
OK thanks for your information, done.